### PR TITLE
config: Add MaxBlockHistoryLookback option

### PR DESCRIPTION
**Pull request by ohill**
## Summary

There is no way to request a node to store more blocks if it is not archival. This adds a MaxBlockHistoryLookback similar to MaxAcctLookback config option.

## Test Plan

No tests yet

### CODE REVIEW

File: ledger/ledger.go [hunk 1]
```diff
@@ -454,6 +454,11 @@ func (l *Ledger) notifyCommit(r basics.Round) basics.Round {
 	}()
 	minToSave := l.trackers.committedUpTo(r)
 
+	// Check if additional block history is configured, and adjust minToSave if so.
+	if configuredMinToSave := r.SubSaturate(basics.Round(l.cfg.MaxBlockHistoryLookback)); configuredMinToSave < minToSave {
```
Comment (winder) [inline ledger/ledger.go:458]: Condition looks to be backwards. Presumably this value already takes into account the `1321` and `MaxAcctLookback` settings, so we need to ensure it doesn't shrink.
```suggestion
	if configuredMinToSave := r.SubSaturate(basics.Round(l.cfg.MaxBlockHistoryLookback)); configuredMinToSave > minToSave {
```

**Review by winder — CHANGES_REQUESTED** [review]

File: ledger/ledger.go (hunk 1 above)
Comment (winder) [reply ledger/ledger.go:458]: I mistook the variable to be "min number of rounds to save" when it is actually "minimum round to save". Looks like this code is correct.

**Review by jannotti — DISMISSED** [review]
Seems good to me.

Do we ever complain about non-sensical combinations of config?  If someone sets this to, say, 3000 but also has Archival set, they are probably confused.

File: config/localTemplate.go [hunk 2]
```diff
@@ -520,6 +520,10 @@ type Local struct {
 	// i.e. the ledger can answer account states questions for the range Latest-MaxAcctLookback...Latest
 	MaxAcctLookback uint64 `version[23]:"4"`
 
+	// BlockHistoryLookback sets the max lookback range for block information.
+	// i.e. the block DB can return transaction IDs for questions for the range Latest-MaxBlockHistoryLookback...Latest
+	MaxBlockHistoryLookback uint64 `version[31]:"0"`
```
[the remark this answers is not in the document]
Comment (algorandskiy) [reply config/localTemplate.go:525]: I think v31 not released yet so it is fine

**Review by algorandskiy — COMMENTED** [review]
the code LGTM but please add a unit test and perform a manual test.

**Review by algorandskiy — APPROVED** [review]
Thank you for the change!

**Review by algorandskiy — APPROVED** [review]

**Review by winder — APPROVED** [review]

**Pull request merged**